Shopify/shopify_app
# Support `ENV["HOST"]` in Content Security Policy

![](https://media1.giphy.com/media/ZTUfoXigKRpCM/giphy.gif?cid=ecf05e47300zo7ghyt4s61mo1j93xlktzrqawom7oyn64y8g&rid=giphy.gif&ct=g)

### What this PR does

Adds `ENV["HOST"]` (if present) to the allowed domains in the Content-Security-Policy header.

### Reviewer's guide to testing

1. Create a new rails app (or use existing rails app)
2. Install this gem in Gemfile:
`gem "shopify_app", git: 'https://github.com/shopify/shopify_app.git', branch: 'nelsonwittwer/csp_host'`
3. Bundle install
4. Ensure you have a `HOST` environment variable in your `.env` file
5. Navigate to embedded app in admin
6. View headers in `/products` call made in home controller
7. Verify your `HOST` env variable is added to content security policy

<img width="1125" alt="Screen Shot 2022-09-12 at 2 57 44 PM" src="https://user-images.githubusercontent.com/758143/189734628-d36cbdf1-b271-4530-80ec-fc4df9b1e50f.png">

### Checklist

Before submitting the PR, please consider if any of the following are needed:
